src/hooks/useMedications.js

```diff
@@ -67,11 +69,32 @@ export default function useMedications(setActivePage, selectedDayString) {
     updateMedicationsDiary(deletedMedicationsDiary)
   }
 
+  function saveCopy() {
+    const copyToDayString = createDateString(copyToDay)
+    const index = medicationsDiary.findIndex(
+      day => day.date === copyToDayString
+    )
+
+    if (index > -1) {
+      updateMedicationsDiary(draft => {
+        draft[index].medications = activeMedications
+      })
+    } else {
+      const addedMedicationsDiary = produce(medicationsDiary, draft => {
+        draft.push({ date: copyToDayString, medications: activeMedications })
+      })
+      updateMedicationsDiary(addedMedicationsDiary)
+    }
+  }
+
   return {
     activeMedications,
     medicationToEditId,
     setMedicationToEditId,
     handleSubmit,
     deleteSingleMedication,
+    copyToDay,
```

Comment:
Very nice code, on the long run, it could be better to put all those operations like copy into an own customHook.